theano/gpuarray/c_code/dnn_gi.c

```diff
@@ -293,7 +303,47 @@ APPLY_SPECIFIC(conv_gi)(PyGpuArrayObject *kerns, PyGpuArrayObject *output,
     cuda_exit(c->ctx);
     return 1;
   }
+  // save worksize for next time/cache
+  prev_algo.wsSize = worksize;
+  
+  // Add to the cache
+  if (params->choose_algo)
+      dnn_conv_update_cache(hashkey, prev_algo);
+    }  
+
+    #ifdef DEBUG
+    char algorithm_name[128];
```

Comment:
`algorithm_name` is already defined earlier with `#idef DEBUG`, so this line should be removed.